tests/scripts/test_scripts.py

```diff
@@ -711,7 +711,7 @@ def test_train_rl_double_normalization(tmpdir: str, rng):
     th.save(net, tmppath)
 
     log_dir_data = os.path.join(tmpdir, "train_rl")
-    with pytest.warns(RuntimeWarning):
+    with pytest.warns(RuntimeWarning) as record:
```

Comment:
Thoughts on just:

with pytest.warns(RuntimeWarning, match=r"Applying normalization to already normalized reward function.*"):

?

The upside is a little less code to think about and if the underlying code starts to generate other warnings as well we don't need to change the test.

The main downside seems to be that we're not longer guaranteeing that this is the only warning surfaced, but I'm not sure that maintaining that guarantee is going to be easy given the number of warnings I see elsewhere in the codebase due to usage of deprecated functions.

Reply:
I prefer that, actually, little bit easier to reason about and cleaner.

The change I’ve implemented actually has that same flaw I think, since it filters the list of warnings just to the warnings that contain this message and assets that the filtered list has length 1. Both solutions are robust to other warnings being generated I think.

Reply:
Done.